sunflower-land/sunflower-land
# [FEAT] Error modals for auth machine

# Description

This PR adds distinct error modals to the onboarding flow. The aim is to help the user navigate through easily and when something goes wrong to give them enough feedback so be able to fix the problem.

I have included error modals for:
- Wrong chain
- No web 3
- Rejected sign transaction
- General error modal for all other connecting errors (as more become apparent we can add other error states)

<img width="1027" alt="Screen Shot 2022-02-13 at 2 00 34 pm" src="https://user-images.githubusercontent.com/17863697/153736569-81fed641-e94a-4171-83b5-559041e0ef39.png">
<img width="737" alt="Screen Shot 2022-02-13 at 2 01 44 pm" src="https://user-images.githubusercontent.com/17863697/153736642-5adea6e2-0415-48c4-978e-c839ad9e911b.png">


https://user-images.githubusercontent.com/17863697/153736649-0ba26bc0-9d65-4a9e-8379-7817a92b3f28.mov



https://user-images.githubusercontent.com/17863697/153736661-b9d9e12a-28ce-447e-905b-7e34fc124eda.mov


Fixes (#issue)

## Type of change

Please delete options that are not relevant.


# How Has This Been Tested?

- Load game in incognito (No Web3)
- Test link to docs work
- Load in normal browser. Reject Sign Tx
- Check link to Terms of Service work. Click try again.
- Load game connected to another chain. 
- Check link to polygon set up docs.

# Checklist:
